lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js

```diff
@@ -45,16 +46,19 @@ export class LhcPeriodsOverviewModel extends Observable {
         this._registerFilter(this._namesFilterModel);
         this._yearsFilterModel = new TextTokensFilterModel();
         this._registerFilter(this._yearsFilterModel);
+        this._beamTypesFilterModel = new TextTokensFilterModel();
+        this._registerFilter(this._beamTypesFilterModel);
     }
 
     /**
      * Fetch all the relevant LHC Periods from the API
+     * @param {boolean} clear remove fetched data and force to set loading state before fetching once again
      * @returns {void}
      */
-    async fetchLhcPeriods() {
-        const keepExisting = this._pagination.currentPage > 1 && this._pagination.isInfiniteScrollEnabled;
+    async fetchLhcPeriods(clear = false) {
+        const keepExisting = !clear && this._pagination.currentPage > 1 && this._pagination.isInfiniteScrollEnabled;
 
-        if (!this._pagination.isInfiniteScrollEnabled) {
+        if (keepExisting) {
```

Comment:
No no no!
First: you don't set loading state if you keep existing, it's the other way around.
Then: a fetch function should not accept a parameter named `clear`, the choice of clearing or not should come before! (actually, fetch should not even set the data, this should be done in an `loadCurrentPage` or something similar, but this is not the point here)

Reply:
fixed